core/macros/src/main/scala/io/udash/macros/PropertyMacros.scala

```diff
@@ -336,16 +336,17 @@ class PropertyMacros(val ctx: blackbox.Context) extends AbstractMacroCommons(ctx
 
     def genTree(source: List[(Select, TermName)], targetTree: Tree): Tree = source match {
       case (select, term) :: _ if select.tpe.typeConstructor =:= SeqTpe.typeConstructor =>
-        q"""$targetTree.getSubSeq[${select.tpe.typeArgs.head.widen}](${q"_.$term"}, ${term.decodedName.toString})"""
+        val widenTpe = select.tpe.typeArgs.head.widen
+        q"""$targetTree.getSubSeq[$widenTpe](${q"_.$term"}, ${term.decodedName.toString})"""
       case (select, term) :: Nil if hasModelPropertyCreator(select.tpe.widen) =>
-        q"""{
-            val tmp = $targetTree
-            tmp.getSubModel[${select.tpe.widen}](${q"_.$term"}, ${term.decodedName.toString})
-        }"""
+        val widenTpe = select.tpe.widen
+        q"""$targetTree.getSubModel[$widenTpe](${q"_.$term"}, ${term.decodedName.toString})"""
       case (select, term) :: tail if hasModelPropertyCreator(select.tpe.widen) =>
-        genTree(tail, q"""$targetTree.getSubModel[${select.tpe.widen}](${q"_.$term"}, ${term.decodedName.toString}).asInstanceOf[$ModelPropertyMacroApiCls[${select.tpe.widen}]]""")
+        val widenTpe = select.tpe.widen
+        genTree(tail, q"""$targetTree.getSubModel[$widenTpe](${q"_.$term"}, ${term.decodedName.toString}).asInstanceOf[$ModelPropertyMacroApiCls[$widenTpe]]""")
       case (select, term) :: _ =>
-        q"""$targetTree.getSubProperty[${select.tpe.widen}](${q"_.$term"}, ${term.decodedName.toString})"""
+        val widenTpe = select.tpe.widen
+        q"""$targetTree.getSubProperty[$widenTpe](${q"_.$term.asInstanceOf[$widenTpe]"}, ${term.decodedName.toString})"""
```

Comment:
Ok, I played with this a but trying to understand what _actually_ is going on here and I came to a very professional conclusion that `select.tpe` returns some kind of _bullshit_ type with an unbound wildcard that I have no idea where it comes from.

However, a more "principled" solution than casting might be to obtain less broken type. Instead of computing `List[(Select, TermName)]` you could compute `List[(Type, Symbol)]` where the type is the type of the prefix on which the symbol is selected. The type of selection itself could be computed as `symbol.typeSignatureIn(prefixTpe)`.

Reply:
That's a quick proposition on how to compute `List[(Type, Symbol)]`, although I haven't tested it:

```scala
    def parsePath(tree: Tree): List[(Type, Symbol)] = {
      def symbolPath(t: Tree): List[Symbol] = t match {
        case Select(pre, _) => t.symbol :: symbolPath(pre)
        case _ => Nil
      }
      def mkPath(prefix: Type, syms: List[Symbol]): List[(Type, Symbol)] = syms match {
        case Nil => Nil
        case head :: tail => (prefix, head) :: mkPath(head.typeSignatureIn(prefix), tail)
      }
      mkPath(weakTypeOf[A], symbolPath(tree))
    }
```